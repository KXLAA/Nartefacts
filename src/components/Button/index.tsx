import * as S from './styles'
import { capitalize } from 'utils/utils'

export type ButtonProps = {
  text?: string
  buttonColor?: string
  textColor?: string
}

export const Button: React.FC<ButtonProps> = ({
  buttonColor,
  text,
  textColor,
}) => {
  return (
    <S.Button buttonColor={buttonColor} textColor={textColor} role="button">
      {capitalize(text!) || 'Button'}
    </S.Button>
  )
}
