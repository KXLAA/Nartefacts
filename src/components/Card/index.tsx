import Image from 'next/image'

import * as S from '@/components/Card/styles'

export type CardProps = {
  image?: string | null
  colors?: string[] | null
  likeCount?: number | null
  textColor?: string | null
  albumArt?: string | null
  spotify?: string | null
  apple?: string | null
  type?: string | null
  description?: string | null
  title?: string | null
  artist?: {
    name?: string | null
    photoUrl?: string | null
    biography?: string | null
  } | null
}

export const Card: React.FC<CardProps> = ({ albumArt }) => {
  return (
    <S.Card>
      <S.Gradient />
      <S.ImageWrapper>
        <Image
          src={albumArt!}
          height={385}
          width={385}
          alt={'album art'}
          layout="responsive"
        />
      </S.ImageWrapper>
    </S.Card>
  )
}
