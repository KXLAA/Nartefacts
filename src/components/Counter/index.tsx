import * as S from './styles'

export type CounterProps = {
  count?: number | undefined | null
}

export const Counter: React.FC<CounterProps> = ({ count }) => {
  return (
    <S.Wrapper>
      <S.Counter>
        <p>{`${count} Gradients Generated`} </p>
      </S.Counter>
    </S.Wrapper>
  )
}
