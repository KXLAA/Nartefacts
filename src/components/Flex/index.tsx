import { styled } from '@/lib/stitches.config'

export const Flex = styled('div', {
  boxSizing: 'border-box',
  display: 'flex',
  variants: {
    direction: {
      row: {
        flexDirection: 'row',
      },
      column: {
        flexDirection: 'column',
      },
      rowReverse: {
        flexDirection: 'row-reverse',
      },
      columnReverse: {
        flexDirection: 'column-reverse',
      },
    },
    align: {
      start: {
        alignItems: 'flex-start',
      },
      center: {
        alignItems: 'center',
      },
      end: {
        alignItems: 'flex-end',
      },
      stretch: {
        alignItems: 'stretch',
      },
      baseline: {
        alignItems: 'baseline',
      },
    },
    justify: {
      start: {
        justifyContent: 'flex-start',
      },
      center: {
        justifyContent: 'center',
      },
      end: {
        justifyContent: 'flex-end',
      },
      between: {
        justifyContent: 'space-between',
      },
    },
    wrap: {
      noWrap: {
        flexWrap: 'nowrap',
      },
      wrap: {
        flexWrap: 'wrap',
      },
      wrapReverse: {
        flexWrap: 'wrap-reverse',
      },
    },
    gap: {
      1: {
        gap: '$1',
      },
      2: {
        gap: '$2',
      },
      3: {
        gap: '$3',
      },
      4: {
        gap: '$4',
      },
      5: {
        gap: '$5',
      },
      6: {
        gap: '$6',
      },
      8: {
        gap: '$8',
      },
      10: {
        gap: '$10',
      },
      12: {
        gap: '$12',
      },
      16: {
        gap: '$16',
      },
      20: {
        gap: '$20',
      },
      24: {
        gap: '$24',
      },
      32: {
        gap: '$32',
      },
      40: {
        gap: '$40',
      },
      48: {
        gap: '$48',
      },
      56: {
        gap: '$56',
      },
      64: {
        gap: '$64',
      },
    },
  },
  defaultVariants: {
    direction: 'row',
    align: 'stretch',
    justify: 'start',
    wrap: 'noWrap',
  },
})
