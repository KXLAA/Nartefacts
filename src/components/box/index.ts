export { Box } from './Box'
export type { BoxProps } from './box.types'
