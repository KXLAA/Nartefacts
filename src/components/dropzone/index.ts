export { Dropzone } from './Dropzone'
export type { UploadState } from './types'
