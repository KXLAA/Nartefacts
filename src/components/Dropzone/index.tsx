import { useCallback } from 'react'
import { useDropzone } from 'react-dropzone'
import * as S from './styles'
import { UploadPlus } from '../Icons'
import { useS3Upload } from 'next-s3-upload'
import {
  useGenerateColorsMutation,
  useUpdateAnalyticsMutation,
} from 'graphql/generated/graphql'
import { RotatingLines } from 'react-loader-spinner'
import { colorsTuple } from 'components/Palette'
import { UploadState } from 'pages/create'

export type DropzoneProps = {
  upload: UploadState
  setUpload: React.Dispatch<React.SetStateAction<UploadState>>
}

export const Dropzone: React.FC<DropzoneProps> = ({ upload, setUpload }) => {
  const { uploadToS3 } = useS3Upload()
  const [generateColors] = useGenerateColorsMutation()
  const [updateAnalytics] = useUpdateAnalyticsMutation()

  const onDrop = useCallback(async (acceptedFiles) => {
    setUpload((prev) => ({ ...prev, isUploading: true }))
    try {
      const { url } = await uploadToS3(acceptedFiles[0])
      const id = '9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d'
      setUpload((prev) => ({ ...prev, imageUrl: url }))
      if (url) {
        setUpload((prev) => ({ ...prev, isUploading: false }))
        const { data } = await generateColors({
          variables: {
            imageUrl: url,
          },
        })
        await updateAnalytics({
          variables: {
            id: id,
          },
          optimisticResponse: {
            updateAnalytics: {
              id: id,
              generatedPalettes: 1,
            },
          },
        })
        setUpload((prev) => ({
          ...prev,
          colors: data?.generateColors?.colors as colorsTuple,
        }))
      }
    } catch (error: unknown) {
      if (error instanceof Error) {
        setUpload((prev) => ({ ...prev, error: (error as Error).message }))
      }
    }

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  const { getRootProps, getInputProps, isDragActive } = useDropzone({
    onDrop,
    maxFiles: 1,
    accept: 'image/jpeg,image/png',
  })

  return (
    <S.Drop {...getRootProps()}>
      <input {...getInputProps()} />
      {!upload.error ? (
        <S.Content>
          {upload.isUploading ? (
            <RotatingLines width="100" strokeColor="#FF5733" />
          ) : (
            <>
              <UploadPlus size={72} strokeWidth={0.5} color="#5A5A5A" />

              {isDragActive ? (
                <p>drop the files here ...</p>
              ) : (
                <p>drag {'n'} drop your image here, or click to select files</p>
              )}
            </>
          )}
        </S.Content>
      ) : (
        <p>{upload.error}</p>
      )}
    </S.Drop>
  )
}
