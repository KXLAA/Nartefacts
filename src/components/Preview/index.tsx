import { useState } from 'react'
import toast from 'react-hot-toast'

import { Button } from '@/components/Button'
import { Flex } from '@/components/Flex'
import { Generated } from '@/components/Generated'
import { ColorsTuple } from '@/components/Palette'
import * as S from '@/components/Preview/styles'
import { Spacer } from '@/components/Spacer'
import { Toast } from '@/components/Toast'
import { useCreatedStore } from '@/lib/store'

export type PreviewProps = {
  heading?: string
  imageUrl: string
  reset?: () => void
  colors: ColorsTuple
}

export const Preview: React.FC<PreviewProps> = ({
  imageUrl,
  colors,
  reset,
}) => {
  const store = useCreatedStore()
  const [disable, setDisable] = useState<boolean>(!colors)

  const save = () => {
    setDisable(true)
    if (colors && imageUrl) store.addGeneratedColor(imageUrl, colors)
    toast(() => <Toast imageUrl={imageUrl} />)
  }

  return (
    <S.Wrapper data-testid="Preview">
      <Flex gap={4}>
        <Button label="Save" disabled={disable} onClick={save} fullWidth />

        <Button
          variant="dark"
          label="Refresh"
          onClick={reset ? reset : undefined}
          fullWidth
        />
      </Flex>
      <Spacer horizontal={4} />
      <Generated imageUrl={imageUrl} colors={colors} />
    </S.Wrapper>
  )
}
