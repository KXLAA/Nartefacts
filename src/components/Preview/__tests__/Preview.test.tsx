import { render } from '@testing-library/react'
import { imageUrl, colors } from 'components/utils'
import { Preview } from '..'
import userEvent from '@testing-library/user-event'
import '../../../../matchMedia.mock'

describe('<Preview />', () => {
  it('should render the component', () => {
    const { getByTestId, getByAltText } = render(
      <Preview colors={colors} imageUrl={imageUrl} />,
    )
    const preview = getByTestId(/preview/i)
    const image = getByAltText('user uploaded image')
    expect(image).toBeInTheDocument()
    expect(image).toBeTruthy()
    expect(preview).toBeInTheDocument()
  })

  it('should render the component correctly', () => {
    const { container } = render(
      <Preview colors={colors} imageUrl={imageUrl} />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('clicking the save button should save the image and disable the save button', async () => {
    const user = userEvent.setup()
    const { getByText } = render(
      <Preview colors={colors} imageUrl={imageUrl} />,
    )
    const saveButton = getByText(/save/i)
    expect(saveButton).not.toBeDisabled()

    await user.click(saveButton)
    const toast = getByText(/saved/i)
    expect(toast).toBeInTheDocument()
    expect(saveButton).toBeDisabled()
  })
})
