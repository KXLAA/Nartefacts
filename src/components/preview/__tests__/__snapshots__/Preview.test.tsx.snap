// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preview /> should render <Preview /> correctly 1`] = `
<div
  class="nartefacts-c-PJLV nartefacts-c-ODbym"
  data-testid="Preview"
>
  <div
    class="nartefacts-c-PJLV nartefacts-c-fZrNKq nartefacts-c-fZrNKq-ejCoEP-direction-row nartefacts-c-fZrNKq-iSZeJd-gap-4"
  >
    <button
      class="nartefacts-c-lchVdd nartefacts-c-lchVdd-gHFwzb-size-lg nartefacts-c-lchVdd-eDnYwL-variant-light nartefacts-c-lchVdd-cFpcyF-fullWidth-true nartefacts-c-lchVdd-iPJLV-css"
    >
      Save
    </button>
    <button
      class="nartefacts-c-lchVdd nartefacts-c-lchVdd-gHFwzb-size-lg nartefacts-c-lchVdd-eVtbBr-variant-dark nartefacts-c-lchVdd-cFpcyF-fullWidth-true nartefacts-c-lchVdd-iPJLV-css"
    >
      Refresh
    </button>
  </div>
  <span
    aria-hidden="true"
    class="nartefacts-c-hakyQ nartefacts-c-hakyQ-jbdhzH-width-4 nartefacts-c-hakyQ-bbciZP-height-4"
  />
  <div>
    <div
      class="nartefacts-c-iYrbGo"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
        />
        <img
          alt="user uploaded image"
          data-nimg="responsive"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <span
      aria-hidden="true"
      class="nartefacts-c-hakyQ nartefacts-c-hakyQ-jbdhzH-width-4 nartefacts-c-hakyQ-bbciZP-height-4"
    />
    <div
      class="nartefacts-c-PJLV nartefacts-c-dwnvnm nartefacts-c-dwnvnm-gyNLch-columns-4 nartefacts-c-dwnvnm-kdofoX-gap-2"
    >
      <div
        class="nartefacts-c-PJLV nartefacts-c-irykEI nartefacts-c-PJLV-idrbDvt-css"
        data-testid="color-box"
      >
        <span />
      </div>
      <div
        class="nartefacts-c-PJLV nartefacts-c-irykEI nartefacts-c-PJLV-ieaNIii-css"
        data-testid="color-box"
      >
        <span />
      </div>
      <div
        class="nartefacts-c-PJLV nartefacts-c-irykEI nartefacts-c-PJLV-igbSeBM-css"
        data-testid="color-box"
      >
        <span />
      </div>
      <div
        class="nartefacts-c-PJLV nartefacts-c-irykEI nartefacts-c-PJLV-igKYXrQ-css"
        data-testid="color-box"
      >
        <span />
      </div>
      <div
        class="nartefacts-c-PJLV nartefacts-c-irykEI nartefacts-c-PJLV-ilbfIFJ-css"
        data-testid="color-box"
      >
        <span />
      </div>
      <div
        class="nartefacts-c-PJLV nartefacts-c-irykEI nartefacts-c-PJLV-iecSHAT-css"
        data-testid="color-box"
      >
        <span />
      </div>
      <div
        class="nartefacts-c-PJLV nartefacts-c-irykEI nartefacts-c-PJLV-iljWqZn-css"
        data-testid="color-box"
      >
        <span />
      </div>
      <div
        class="nartefacts-c-PJLV nartefacts-c-irykEI nartefacts-c-PJLV-ifCAGtg-css"
        data-testid="color-box"
      >
        <span />
      </div>
    </div>
  </div>
</div>
`;
